src/components/HappinessCalendar.tsx

```diff
@@ -105,6 +105,7 @@ export default function HappinessCalendar({
                 <DayCell
                   happiness={matchingHappiness}
                   isSelected={
+                    selectedEntry !== undefined &&
```

Comment:
this line is unnecessary, just the selectedEntry && will short circuit if selectedEntry is undefined.

Reply:
I get TypeScript errors if I remove that statement, so that's why I added it.

Reply:
![image](https://github.com/jonathanjma/happiness-app-frontend/assets/67171896/d36488c7-98c6-4dd4-a595-863f60ee67fb)

Reply:
oh ok valid. yeah I was just thinking for conditional rendering for some reason but I guess when you pass in a boolean you do need it.